resources/qml/Settings/SettingView.qml

```diff
@@ -9,229 +9,406 @@ import QtQuick.Layouts 1.1
-            //In addition, while it works for 5.5 and higher, the ordering of the actual combo box drop down changes,
-            //causing nasty issues when selecting different options. So disable asynchronous loading of enum type completely.
-            asynchronous: model.type != "enum" && model.type != "extruder"
-            active: model.type != undefined
+        anchors
+        {
+            top: parent.top
+            left: parent.left
+            leftMargin: UM.Theme.getSize("default_margin").width
+            right: parent.right
+            rightMargin: UM.Theme.getSize("default_margin").width
+        }
+        height: visible ? UM.Theme.getSize("setting_control").height : 0
+        Behavior on height { NumberAnimation { duration: 100 } }
 
-            source:
+        TextField
+        {
+            id: filter;
+            visible: !monitoringPrint
+
+            anchors.left: parent.left
+            anchors.right: clearFilterButton.left
+            anchors.rightMargin: UM.Theme.getSize("default_margin").width
+
+            placeholderText: catalog.i18nc("@label:textbox", "Filter...")
+
+            style: TextFieldStyle
             {
-                switch(model.type)
-                {
-                    case "int":
-                        return "SettingTextField.qml"
-                    case "float":
-                        return "SettingTextField.qml"
-                    case "enum":
-                        return "SettingComboBox.qml"
-                    case "extruder":
-                        return "SettingExtruder.qml"
-                    case "bool":
-                        return "SettingCheckBox.qml"
-                    case "str":
-                        return "SettingTextField.qml"
-                    case "category":
-                        return "SettingCategory.qml"
-                    default:
-                        return "SettingUnknown.qml"
-                }
+                textColor: UM.Theme.getColor("setting_control_text");
+                font: UM.Theme.getFont("default");
+                background: Item {}
             }
 
-            // Binding to ensure that the right containerstack ID is set for the provider.
-            // This ensures that if a setting has a limit_to_extruder id (for instance; Support speed points to the
-            // extruder that actually prints the support, as that is the setting we need to use to calculate the value)
-            Binding
+            property var expandedCategories
+            property bool lastFindingSettings: false
+
+            onTextChanged:
             {
-                target: provider
-                property: "containerStackId"
-                when: model.settable_per_extruder || (inheritStackProvider.properties.limit_to_extruder != null && inheritStackProvider.properties.limit_to_extruder >= 0);
-                value:
+                definitionsModel.filter = {"label": "*" + text};
```

Comment:
I think this would also continue to search in English even if you set the language differently. That is a change that would need to be put through to Uranium though, in the way that this filter is applied.